tests/executor_test.py

```diff
@@ -43,6 +44,39 @@ def test_executor(self):
         step_graph = StepGraph.from_params({"sum": {"type": "sum_numbers", "a": 1, "b": 2}})
         executor = Executor(workspace)
         assert len(executor.workspace.step_cache) == 0
-        executor.execute_step_graph(step_graph)
+        output = executor.execute_step_graph(step_graph)
+        assert "sum" in output.successful
         assert len(executor.workspace.step_cache) == 1
         assert executor.workspace.step_cache[step] == 3
+
+    def test_executor_with_failing_steps(self):
+        workspace = LocalWorkspace(self.TEST_DIR)
+        step_graph = StepGraph.from_params(
+            {
+                "successful_step": {
+                    "type": "sleep-print-maybe-fail",
+                    "string": "This ran perfectly.",
+                    "seconds": 0,
+                    "fail": False,
+                },
+                "failing_step": {
+                    "type": "sleep-print-maybe-fail",
+                    "string": "This should fail.",
+                    "seconds": 0,
+                    "fail": True,
+                },
+                "dependent_step": {
+                    "type": "sleep-print-maybe-fail",
+                    "string": {"type": "ref", "ref": "failing_step"},
+                    "seconds": 0,
+                    "fail": False,
+                },
+            }
+        )
+        executor = Executor(workspace)
+        assert len(executor.workspace.step_cache) == 0
+        output = executor.execute_step_graph(step_graph)
```

Comment:
When looking at it like this, this seems a little dangerous. It's easy to accidentally write code the swallows errors. It should be very clear in the docs that `execute_step_graph()` will not propagate exceptions.